lightningd/offer.c

```diff
@@ -231,6 +231,48 @@ static const struct json_command disableoffer_command = {
 };
 AUTODATA(json_command, &disableoffer_command);
 
+static struct command_result *json_enableoffer(struct command *cmd,
+						const char *buffer,
+						const jsmntok_t *obj UNNEEDED,
+						const jsmntok_t *params)
+{
+	struct json_stream *response;
+	struct sha256 *offer_id;
+	struct wallet *wallet = cmd->ld->wallet;
+	const char *b12;
+	const struct json_escape *label;
+	enum offer_status status;
+
+	if (!param_check(cmd, buffer, params,
+			 p_req("offer_id", param_sha256, &offer_id),
+			 NULL))
+		return command_param_failed();
+
+	b12 = wallet_offer_find(tmpctx, wallet, offer_id, &label, &status);
+	if (!b12)
+		return command_fail(cmd, LIGHTNINGD, "Unknown offer");
+
+	if (offer_status_active(status))
+		return command_fail(cmd, OFFER_ALREADY_DISABLED,
```

Comment:
Just a small detail: `OFFER_ALREADY_DISABLED` code does not reflect what is actually happening in this case.
Maybe the correct code should be `OFFER_ALREADY_EXISTS` or a new one `OFFER_ALREADY_ENABLED`.

Reply:
Good point thanks!